Mathlib/Algebra/BigOperators/Associated.lean

```diff
@@ -50,6 +50,13 @@ theorem exists_mem_finset_dvd {s : Finset β} {f : β → α} : p ∣ s.prod f 
 
 end Prime
 
+theorem Prod.associated_iff {M N : Type*} [Monoid M] [Monoid N] {x z : M × N} :
+    x ~ᵤ z ↔ x.1 ~ᵤ z.1 ∧ x.2 ~ᵤ z.2 :=
+  ⟨fun ⟨u, hu⟩ => ⟨⟨(MulEquiv.prodUnits.toFun u).1, (Prod.eq_iff_fst_eq_snd_eq.1 hu).1⟩,
+    ⟨(MulEquiv.prodUnits.toFun u).2, (Prod.eq_iff_fst_eq_snd_eq.1 hu).2⟩⟩,
+  fun ⟨⟨u₁, h₁⟩, ⟨u₂, h₂⟩⟩ =>
+    ⟨MulEquiv.prodUnits.invFun (u₁, u₂), Prod.eq_iff_fst_eq_snd_eq.2 ⟨h₁, h₂⟩⟩⟩
+
```

Comment:
This seems like a very strange file for this lemma, as it's not about a big operator!

Reply:
The whole story is that I need a file which talks about associated elements and which imports both [Mathlib.Algebra.Associated](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Associated.html) and [Mathlib.Algebra.BigOperators.Associated](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/BigOperators/Associated.html). Do you think I should move the result somewhere else?

Reply:
Why do you need a file that imports Mathlib.Algebra.BigOperators.Associated?

Reply:
I copied and pasted the wrong link. I am sorry. I need a file which imports both [Mathlib.Algebra.Associated](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Associated.html) and [Mathlib.Algebra.Group.Prod](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Group/Prod.html).